app/controllers/users/confirmations_controller.rb

```diff
@@ -48,8 +48,15 @@ def show
       respond_with_navigational(resource) { render :show }
     elsif resource.errors.empty?
       set_official_position if resource.has_official_email?
-      resource.confirm # Last change: confirm happens here for people with passwords instead of af the top of the show action
-      set_flash_message(:notice, :confirmed) if is_flashing_format?
+
+      if resource.confirmed_at 
+        message = :already_confirmed
+      else
+        resource.confirm # Last change: confirm happens here for people with passwords instead of af the top of the show action
+        message = :confirmed
+      end
+     
+      set_flash_message(:notice, message) if is_flashing_format?
```

Comment:
Comment automatically generated by Pronto.

Layout/TrailingWhitespace: Trailing whitespace detected. (https://rubystyle.guide#no-trailing-whitespace)

```suggestion

      set_flash_message(:notice, message) if is_flashing_format?
```